src/sentry/integrations/slack/unlink_identity.py

```diff
@@ -0,0 +1,99 @@
+from django.db import IntegrityError
+from django.http import Http404
+from django.views.decorators.cache import never_cache
+
+from sentry.models import Integration, Identity, IdentityProvider, Organization
+from sentry.utils.http import absolute_uri
+from sentry.utils.signing import sign, unsign
+from sentry.web.decorators import transaction_start
+from sentry.web.frontend.base import BaseView
+from sentry.web.helpers import render_to_response
+from sentry.shared_integrations.exceptions import ApiError
+
+from .client import SlackClient
+from .utils import logger
+
+
+def build_unlinking_url(integration_id, organization_id, slack_id, channel_id, response_url):
+    signed_params = sign(
+        integration_id=integration_id,
+        organization_id=organization_id,
+        slack_id=slack_id,
+        channel_id=channel_id,
+        response_url=response_url,
+    )
+
+    return absolute_uri(
+        reverse("sentry-integration-slack-unlink-identity", kwargs={"signed_params": signed_params})
+    )
+
+
+class SlackUnlinkIdentityView(BaseView):
+    @transaction_start("SlackUnlinkIdentityView")
+    @never_cache
+    def handle(self, request, signed_params):
+        params = unsign(signed_params.encode("ascii", errors="ignore"))
+
+        try:
+            organization = Organization.objects.get(
+                id__in=request.user.get_orgs(), id=params["organization_id"]
+            )
+        except Organization.DoesNotExist:
+            raise Http404
+
+        try:
+            integration = Integration.objects.get(
+                id=params["integration_id"], organizations=organization
+            )
+        except Integration.DoesNotExist:
+            raise Http404
+
+        try:
+            idp = IdentityProvider.objects.get(external_id=integration.external_id, type="slack")
+        except IdentityProvider.DoesNotExist:
+            raise Http404
+
+        if request.method != "POST":
+            return render_to_response(
+                "sentry/auth-unlink-identity.html",
+                request=request,
+                context={"organization": organization, "provider": integration.get_provider()},
+            )
+
+        # Delete the wrong slack identity.
+        try:
+            identity = Identity.objects.get(idp=idp, external_id=params["slack_id"])
+            identity.delete()
+        except IntegrityError as e:
+            logger.error("slack.unlink.integrity-error", extra=e)
+            raise Http404
+
+        payload = {
+            "replace_original": False,
+            "response_type": "ephemeral",
+            "text": "Your Slack identity has been unlinked from your Sentry account.",
+        }
+
+        client = SlackClient()
+        try:
+            client.post(params["response_url"], data=payload, json=True)
+        except ApiError as e:
+            message = six.text_type(e)
+            # If the user took their time to link their slack account, we may no
```

Comment:
Can we move the copied blocks copied from `link_identity.py` to `/src/sentry/integrations/slack/utils.py`?